e2e/test/FaultInjection.cs

```diff
@@ -179,36 +312,33 @@ public static async Task ActivateFaultInjection(Client.TransportType transport,
                 s_log.WriteLine($"{nameof(FaultInjection)}: Waiting for fault injection to be active: {delay}ms");
                 await Task.Delay(delay).ConfigureAwait(false);
 
-                await testOperation(deviceClient, testDevice).ConfigureAwait(false);
+                // TODO: This needs to be changed - for recoverable faults we need to wait some more time for the client to know about fault and then retry
```

Comment:
Please create a GH issue for all // TODO comments (or use one that aggregates remaining work.) Add the ID like `// TODO #nnn`.